openfeature/hook/__init__.py

```diff
@@ -29,6 +29,11 @@ class HookContext:
     client_metadata: typing.Optional[ClientMetadata] = None
     provider_metadata: typing.Optional[Metadata] = None
 
+    def __setattr__(self, key: str, value: typing.Any) -> None:
+        if hasattr(self, key) and key in ("flag_key", "flag_type", "default_value"):
+            raise AttributeError(f"Attribute {key!r} is immutable")
```

Comment:
What's the difference with just `{key}` here?

Reply:
it just adds quotes around it 😄 ex. `Attribute 'flag_key' is immutable`